### PR TITLE
FIX - for issue 1822

### PR DESCRIPTION
**Pull request by schoffelen**
This is a fix for 1822 to make the partialisation option in ft_connectivityanalysis a little bit less messy.

### CODE REVIEW

File: connectivity/ft_connectivity_corr.m [hunk 1]
```diff
@@ -81,14 +80,13 @@
...
-cmplx       = ft_getopt(varargin, 'complex', 'abs');
+hasjack     = ft_getopt(varargin, 'hasjack',  false);
+cmplx       = ft_getopt(varargin, 'complex',  'abs');
 feedback    = ft_getopt(varargin, 'feedback', 'none');
 dimord      = ft_getopt(varargin, 'dimord');
 powindx     = ft_getopt(varargin, 'powindx');
 pownorm     = ft_getopt(varargin, 'pownorm', 0);
-pchanindx   = ft_getopt(varargin, 'pchanindx');
-allchanindx = ft_getopt(varargin, 'allchanindx');
+pchan       = ft_getopt(varargin, 'pchanindx');
```
Comment (robertoostenveld) [inline connectivity/ft_connectivity_corr.m:89]: here the variable is `pchan`, whereas the option is `pchanindx`. In the help you document the option `pchan`

Comment (robertoostenveld) [reply connectivity/ft_connectivity_corr.m:89]: I see further in the code that the two are rather interchangeable. If you did not change the API, I suggest to keep the name `pchanindx`, it is more consistent with other `chanindx` options elsewhere

File: ft_connectivityanalysis.m [hunk 2]
```diff
@@ -221,22 +221,18 @@
...
   case {'coh' 'csd'}
     if ~isempty(cfg.partchannel)
-      if hasrpt && ~hasjack
+      if hasrpt && ~hasjack && ~isfield(data, 'labelcmb')
         ft_warning('partialisation on single trial observations is not supported, removing trial dimension');
         try
           data = ft_checkdata(data, 'datatype', {'freqmvar' 'freq'}, 'cmbstyle', 'fullfast');
           inparam = 'crsspctrm';
-          hasrpt = 0;
+          hasrpt  = contains(data.dimord, 'rpt');
```
Comment (robertoostenveld) [inline ft_connectivityanalysis.m:229]: can you use `getdimord` instead, or possibly the data dimord that has been determined already earlier in the ft_connectivityanalysis code?

File: ft_connectivityanalysis.m [hunk 3]
```diff
@@ -658,17 +649,25 @@
   case 'coh'
     % coherence (unsquared), if cfg.complex = 'imag' imaginary part of coherency
     optarg = {'complex', cfg.complex, 'dimord', data.dimord, 'feedback', cfg.feedback, 'pownorm', normpow, 'hasjack', hasjack};
-    if ~isempty(cfg.pchanindx), optarg = cat(2, optarg, {'pchanindx', cfg.pchanindx, 'allchanindx', cfg.allchanindx}); end
-    if exist('powindx', 'var'), optarg = cat(2, optarg, {'powindx', powindx}); end
+    if ~isempty(cfg.pchanindx), optarg = cat(2, optarg, {'pchanindx', cfg.pchanindx}); end
```
Comment (robertoostenveld) [inline ft_connectivityanalysis.m:652]: here it again is `pchanindx`, which is inconsistent with the change in the low-level function

**Review by robertoostenveld — CHANGES_REQUESTED** [review]
I think there are still a few things that need to be changed, please see the comments

File: connectivity/ft_connectivity_corr.m (hunk 1 above)
Comment (schoffelen) [reply connectivity/ft_connectivity_corr.m]: this has been resolved I think

**Pull request merged**